h/static/scripts/controllers/autosuggest-dropdown-controller.js

```diff
@@ -0,0 +1,454 @@
+'use strict';
+
+const Controller = require('../base/controller');
+const setElementState = require('../util/dom').setElementState;
+
+const ENTER = 13;
+const UP = 38;
+const DOWN = 40;
+
+/**
+ * Controller for adding autosuggest control to a piece of the page
+ */
+class AutosuggestDropdownController extends Controller {
+
+  /*
+   * @typedef {Object} ConfigOptions
+   * @property {Function} renderListItem - called with every item in the list
+   *   after the listReducer function is called. The return value will be the final
+   *   html that is set in the list item DOM.
+   * @property {Function} listReducer - called at initialization, focus of input,
+   *   and input changes made by user. The function will receieve the full list
+   *   and the current value of the input. This is meant to be an pure function
+   *   that will return a filtered list based on the consumer's domain needs.
+   * @properly {Function} onSelect - called once the user has made a selection of an
+   *   item in the autosuggest. It will receive the item selected as the only argument.
+   * @property {Object} [classNames] - this is the enumerated list of class name
+   *   overrides for consumers to customize the UI.
+   *   Possible values: { container, list, item, activeItem, header }
+   * @property {String} [header] - value of the header label at top of suggestions
+   */
+
+  /**
+   * @param {HTMLElement} inputElement that we are responding to in order to provide
```

Comment:
Presumably this needs to be an `HTMLInputElement` rather than just an `HTMLElement`?